### PR TITLE
Setup Eventhubs service

### PR DESCRIPTION
**Pull request by gearama**
Create the wirframe for event hubs service src, in test/ut folders, one test, vcpkg, ci.yml, cmake files etc, 
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [ ] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [ ] Doxygen docs
- [ ] Unit tests
- [ ] No unwanted commits/changes
- [ ] Descriptive title/description
  - [ ] PR is single purpose
  - [ ] Related issue listed
- [ ] Comments in source
- [ ] No typos
- [ ] Update changelog
- [ ] Not work-in-progress
- [ ] External references or docs updated
- [ ] Self review of PR done
- [ ] Any breaking changes?

### CODE REVIEW

File: sdk/messaging/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+# SPDX-License-Identifier: MIT
+
+cmake_minimum_required (VERSION 3.13)
+
+project (azure-messaging LANGUAGES CXX)
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+
+add_subdirectory(azure-messaging-eventhubs)
```
Comment (LarryOsterman) [inline sdk/messaging/CMakeLists.txt:11]: Missing CRLF?

File: CMakeLists.txt [hunk 2]
```diff
@@ -111,6 +111,7 @@ add_subdirectory(sdk/identity)
 add_subdirectory(sdk/keyvault)
 add_subdirectory(sdk/storage)
 add_subdirectory(sdk/template)
+add_subdirectory(sdk/messaging)
```
Comment (LarryOsterman) [inline CMakeLists.txt:114]: DO you want ot make this conditional on BUILD_AMQP?

File: sdk/messaging/azure-messaging-eventhubs/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    azure-messaging-eventhubs
+    MESSAGING_EVENTHUBS
+    "azure/messaging/eventhubs/dll_import_export.hpp"
+  )
+
+az_rtti_setup(
+  azure-security-keyvault-keys
+  MESSAGING_EVENTHUBS
+  "azure/messaging/eventhubs/rtti.hpp"
+)
```
Comment (LarryOsterman) [inline sdk/messaging/azure-messaging-eventhubs/CMakeLists.txt:85]: CRLF at EOF.

File: sdk/messaging/azure-messaging-eventhubs/inc/azure/messaging/eventhubs.hpp [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+// SPDX-License-Identifier: MIT
+
+/**
+ * @brief Includes all public headers from Azure Messaging Event Hubs SDK library.
+ *
+ */
+
+#pragma once
+
+#include "azure/messaging/eventhubs/eventhubs.hpp"
```
Comment (LarryOsterman) [inline sdk/messaging/azure-messaging-eventhubs/inc/azure/messaging/eventhubs.hpp:11]: CRLF at EOF.

File: sdk/messaging/azure-messaging-eventhubs/src/eventhubs.cpp [hunk 5]
```diff
@@ -0,0 +1,7 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <azure/core/exception.hpp>
+#include <azure/core/http/http.hpp>
+#include "azure/messaging/eventhubs.hpp"
+#include "private/package_version.hpp"
```
Comment (LarryOsterman) [inline sdk/messaging/azure-messaging-eventhubs/src/eventhubs.cpp:7]: CRLF at EOF (I love that github makes it trival to spot these :))

File: sdk/messaging/azure-messaging-eventhubs/test-resources.json [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+        {
+          "name": "[concat('default/', variables('containerName'))]",
+          "type": "blobServices/containers",
+          "apiVersion": "[variables('storageApiVersion')]",
+          "dependsOn": ["[variables('storageAccountName')]"]
+        }
+      ]
+    },
+    {
+      "type": "Microsoft.Devices/IotHubs",
```
Comment (LarryOsterman) [inline sdk/messaging/azure-messaging-eventhubs/test-resources.json:135]: Will the eventhubs tests really need IotHubs resources?

**Review by LarryOsterman — APPROVED** [review]
I don't see anything horrible here, but there are a bunch of things that should probably be addressed before this PR is committed.

File: sdk/messaging/CMakeLists.txt (hunk 1 above)
Comment (RickWinter) [inline sdk/messaging/CMakeLists.txt:11]:
```suggestion
add_subdirectory(azure-messaging-eventhubs)

```

File: sdk/messaging/azure-messaging-eventhubs/README.md [hunk 7]
```diff
@@ -1,3 +1,3 @@
-# Generic Service for Samples
+# Azure Messaging EventHub service
```
[the remark this answers is not in the document]
Comment (RickWinter) [reply sdk/messaging/azure-messaging-eventhubs/README.md:1]: Agreed, we have namespaces which do allow us to keep things tidy.  Go had to use messaging in the path b/c in Go the path of the import is important.

I'd prefer sdk/<service>/<package> format.

File: sdk/messaging/azure-messaging-eventhubs/inc/azure/messaging/eventhubs.hpp (hunk 4 above)
Comment (RickWinter) [inline sdk/messaging/azure-messaging-eventhubs/inc/azure/messaging/eventhubs.hpp:11]:
```suggestion
#include "azure/messaging/eventhubs/eventhubs.hpp"

```

File: sdk/messaging/azure-messaging-eventhubs/src/eventhubs.cpp (hunk 5 above)
Comment (RickWinter) [inline sdk/messaging/azure-messaging-eventhubs/src/eventhubs.cpp:7]:
```suggestion
#include "private/package_version.hpp"

```

File: sdk/messaging/azure-messaging-eventhubs/vcpkg/vcpkg.json [hunk 8]
```diff
@@ -0,0 +1,28 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (RickWinter) [inline sdk/messaging/azure-messaging-eventhubs/vcpkg/vcpkg.json:1]: Why does the diffing tool show this as red background?  Is there something special about this copyright?

Comment (gearama) [reply sdk/messaging/azure-messaging-eventhubs/vcpkg/vcpkg.json:1]: nope

File: sdk/messaging/azure-messaging-eventhubs/README.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (gearama) [reply sdk/messaging/azure-messaging-eventhubs/README.md:1]: done

**Comment by gearama** [comment]
@RickWinter comments resolved, can i merge PR?

File: sdk/messaging/azure-messaging-eventhubs/vcpkg/vcpkg.json (hunk 8 above)
Comment (LarryOsterman) [reply sdk/messaging/azure-messaging-eventhubs/vcpkg/vcpkg.json]: github is looking at the # and saying "This is not valid JSON, you might want to fix it".

For <reasons>, the checked in vcpkg.json file has a copyright notice, we strip it when we push the changes to vcpkg.

**Pull request merged**